src/js/libgeo/GeoBasics.js

```diff
@@ -154,13 +161,27 @@ function addElementNoProof(el) {
 
     // Detect unsupported operations or missing or incorrect arguments
     var op = geoOps[el.type];
+    var isSet = false;
+    var getKind = function(el) {
+        return csgeo.csnames[el].kind;
+    };
+
     if (!op) {
         console.error(el);
         console.error("Operation " + el.type + " not implemented yet");
         return null;
     }
     if (op.signature !== "**") {
-        if (op.signature.length !== (el.args ? el.args.length : 0)) {
+        // check for sets
+        if (!Array.isArray(op.signature) && op.signature.charAt(1) === "s") {
```

Comment:
I *think* I had originally chosen the `**` syntax for arbitrary parameters because I wanted `P*` for points and so on. You use `Ps` instead. I think there is some risk of confusion, since `["Ps"]` would bean a single set of points, as returned by a multi-valued operation, while `"Ps"` without brackets would be an arbitrary number of points. Easy to mix up, I think. Furthermore, `P*` allows for a fairly canonical `P+` one day in cases where we at least one element. It also allows for `["P", "P", "P*"]` or similar if we ever have an operaion that takes two distinguished points and a set of other points. What do you think?